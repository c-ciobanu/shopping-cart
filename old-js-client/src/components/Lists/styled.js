import styled from "styled-components";
import { Link } from "@reach/router";

export const StyledList = styled.li`
	border: 1px solid lightgrey;
	padding: 10px;
	display: grid;
	grid-template-columns: 1fr auto;
	grid-column-gap: 10px;
`;

export const StyledLink = styled(Link)`
	color: inherit;
	text-decoration: none;
`;
