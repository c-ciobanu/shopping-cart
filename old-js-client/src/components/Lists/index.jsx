import React from "react";
import PropTypes from "prop-types";

import ContextMenu from "components/ContextMenu";
import Stack from "components/Stack";

import { StyledList, StyledLink } from "./styled";

function Lists({ options, onDelete }) {
	if (!options.length) {
		return null;
	}

	return (
		<Stack space="sm" component="ul">
			{options.map((option) => (
				<StyledList key={`list-${option.id}`}>
					<StyledLink to={option.link}>{option.name}</StyledLink>

					<ContextMenu options={[{ text: "Delete", action: () => onDelete(option) }]} />
				</StyledList>
			))}
		</Stack>
	);
}

Lists.propTypes = {
	options: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.number.isRequired,
			name: PropTypes.string.isRequired,
			link: PropTypes.string.isRequired
		})
	).isRequired,
	onDelete: PropTypes.func.isRequired
};

export default Lists;
