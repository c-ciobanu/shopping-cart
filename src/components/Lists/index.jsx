import React from "react";
import PropTypes from "prop-types";

import ContextMenu from "components/ContextMenu";

import { StyledList, StyledLink } from "./styled";

function Lists({ options, onDelete }) {
	return (
		<div>
			{options.map((option) => (
				<StyledList key={`list-${option.id}`}>
					<StyledLink to={option.link}>{option.name}</StyledLink>

					<ContextMenu options={[{ text: "Delete", action: () => onDelete(option) }]} />
				</StyledList>
			))}
		</div>
	);
}

Lists.propTypes = {
	options: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.number.isRequired,
			name: PropTypes.string.isRequired,
			link: PropTypes.string.isRequired
		})
	).isRequired,
	onDelete: PropTypes.func.isRequired
};

export default Lists;
