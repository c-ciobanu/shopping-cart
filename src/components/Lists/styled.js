import styled from "styled-components";
import { Link } from "@reach/router";

export const StyledList = styled.div`
	border: 1px solid lightgrey;
	margin-bottom: 10px;
	padding: 10px;
	display: grid;
	grid-template-columns: 1fr auto;
	grid-column-gap: 10px;

	&:last-child {
		margin-bottom: 0;
	}
`;

export const StyledLink = styled(Link)`
	color: inherit;
	text-decoration: none;
`;
