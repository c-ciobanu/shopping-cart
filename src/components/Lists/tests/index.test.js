import React from "react";
import { render, fireEvent } from "@testing-library/react";

import Lists from "../index";

describe("<Lists />", () => {
	it("Expect to not log errors in console", () => {
		const spy = jest.spyOn(global.console, "error");
		render(<Lists options={[]} onDelete={() => {}} />);
		expect(spy).not.toHaveBeenCalled();
	});

	it("Will call on delete function on the 'Delete' context menu voice click", () => {
		const callback = jest.fn();
		const options = ["Option 1", "Option 2", "Option 3"];

		const { container, getByText } = render(<Lists options={options} onDelete={callback} />);

		const moreButtons = container.querySelectorAll("button");

		fireEvent.click(moreButtons[0]);
		fireEvent.click(getByText("Delete"));

		expect(callback).toHaveBeenCalledTimes(1);
		expect(callback).toHaveBeenCalledWith(options[0]);
	});

	it("Should render and match the snapshot", () => {
		const {
			container: { firstChild }
		} = render(<Lists options={[]} onDelete={() => {}} />);
		expect(firstChild).toMatchSnapshot();
	});
});
