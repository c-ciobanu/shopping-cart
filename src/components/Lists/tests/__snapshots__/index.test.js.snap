// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lists /> Should render and match the snapshot 1`] = `
<div>
  <div
    class="sc-AykKG hFqUOC"
  >
    <a
      class="sc-AykKH iYBAuN"
      href="/lists/1"
    >
      Option 1
    </a>
    <div
      class="sc-AykKC fnZNYC"
    >
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="downshift-7-label downshift-7-toggle-button"
        class="sc-AykKF gvJIvP"
        id="downshift-7-toggle-button"
      >
        <svg>
          more.svg
        </svg>
      </button>
      <ul
        aria-labelledby="downshift-7-label"
        class="sc-AykKD hcHMvH"
        id="downshift-7-menu"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
  <div
    class="sc-AykKG hFqUOC"
  >
    <a
      class="sc-AykKH iYBAuN"
      href="/lists/2"
    >
      Option 2
    </a>
    <div
      class="sc-AykKC fnZNYC"
    >
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="downshift-8-label downshift-8-toggle-button"
        class="sc-AykKF gvJIvP"
        id="downshift-8-toggle-button"
      >
        <svg>
          more.svg
        </svg>
      </button>
      <ul
        aria-labelledby="downshift-8-label"
        class="sc-AykKD hcHMvH"
        id="downshift-8-menu"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
  <div
    class="sc-AykKG hFqUOC"
  >
    <a
      class="sc-AykKH iYBAuN"
      href="/lists/3"
    >
      Option 3
    </a>
    <div
      class="sc-AykKC fnZNYC"
    >
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="downshift-9-label downshift-9-toggle-button"
        class="sc-AykKF gvJIvP"
        id="downshift-9-toggle-button"
      >
        <svg>
          more.svg
        </svg>
      </button>
      <ul
        aria-labelledby="downshift-9-label"
        class="sc-AykKD hcHMvH"
        id="downshift-9-menu"
        role="listbox"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;
